kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -881,63 +881,79 @@ protected void handleProduceRequest(KafkaHeaderAndRequest produceHar,
                     errors -> addPartitionResponse.accept(topicPartition, new PartitionResponse(errors));
             final Consumer<Throwable> exceptionConsumer =
                     e -> addPartitionResponse.accept(topicPartition, new PartitionResponse(Errors.forException(e)));
-
             final String fullPartitionName = KopTopic.toString(topicPartition);
 
-            // check KOP inner topic
-            if (isOffsetTopic(fullPartitionName) || isTransactionTopic(fullPartitionName)) {
-                log.error("[{}] Request {}: not support produce message to inner topic. topic: {}",
-                        ctx.channel(), produceHar.getHeader(), topicPartition);
-                errorsConsumer.accept(Errors.INVALID_TOPIC_EXCEPTION);
-                return;
-            }
+            authorize(AclOperation.WRITE, Resource.of(ResourceType.TOPIC, fullPartitionName))
+                    .whenComplete((isAuthorized, ex) -> {
+                        if (ex != null) {
+                            log.error("Write topic authorize failed, topic - {}. {}",
+                                    fullPartitionName, ex.getMessage());
+                            errorsConsumer.accept(Errors.TOPIC_AUTHORIZATION_FAILED);
+                            return;
+                        }
+                        if (!isAuthorized) {
+                            errorsConsumer.accept(Errors.TOPIC_AUTHORIZATION_FAILED);
+                            return;
+                        }
 
```

Comment:
Similar to my previous suggestion, could you extract the followed logic into a single method  `handlePartitionProduce`?